pkg/controller/remotemachineset/awsactuator_test.go

```diff
@@ -187,6 +188,59 @@ func TestAWSActuator(t *testing.T) {
 				Reason: "NoSubnetForAvailabilityZone",
 			},
 		},
+		{
+			name:              "supported spot market options",
+			clusterDeployment: withClusterVersion(testClusterDeployment(), "4.5.0"),
+			poolName:          testMachinePool().Name,
+			existing: []runtime.Object{
+				withSpotMarketOptions(testMachinePool()),
+			},
+			mockAWSClient: func(client *mockaws.MockClient) {
+				mockDescribeAvailabilityZones(client, []string{"zone1"})
+			},
+			expectedMachineSetReplicas: map[string]int64{
+				generateAWSMachineSetName("zone1"): 3,
+			},
+		},
+		{
+			name:              "unsupported spot market options",
+			clusterDeployment: withClusterVersion(testClusterDeployment(), "4.4.0"),
+			poolName:          testMachinePool().Name,
+			existing: []runtime.Object{
+				withSpotMarketOptions(testMachinePool()),
+			},
+			expectedCondition: &hivev1.MachinePoolCondition{
+				Type:   hivev1.UnsupportedConfigurationMachinePoolCondition,
+				Status: corev1.ConditionTrue,
+				Reason: "UnsupportedSpotMarketOptions",
+			},
+		},
+		{
+			name:              "unsupported configuration condition cleared",
+			clusterDeployment: withClusterVersion(testClusterDeployment(), "4.4.0"),
```

Comment:
I did not understand how unsupported configuration condition will be cleared for a cluster with version 4.4.0. Aren't we supporting 4.5.0 and above?

Reply:
It is cleared because the MachinePool is not using spot market options.